File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/Codes1.1/CFD2D/ForwardStepIC2D.m
function Q = ForwardStepIC2D(x2d, y2d, time)
 
% function Q = ForwardStepIC2D(x2d, y2d, time)
% Purpose: compute plane flow configuration 

Globals2D;

gamma = 1.4;

rho = gamma*ones(Np,K);  p = ones(Np,K); 

% M = |u|/c,  c = sqrt(gamma*p/rho)
rhou = rho.*(3*ones(Np, K)); rhov = zeros(Np, K); 
Ener = p/(gamma-1.0) + (rhou.^2+rhov.^2)./(2*rho);

Q(:,:,1) = rho; Q(:,:,2) = rhou; Q(:,:,3) = rhov; Q(:,:,4) = Ener; 
return

File: Nodal_Discontinuous_Galerkin_Methods_Hesthaven/nudg++/trunk/Src/Examples2D/EulerShock2D/ForwardStepIC2D.cpp
// ForwardStepIC2D.m
// function Q = ForwardStepIC2D(x2d, y2d, time)
// 2007/06/30
//---------------------------------------------------------
#include "NDGLib_headers.h"
#include "EulerShock2D.h"


//---------------------------------------------------------
void EulerShock2D::ForwardStepIC2D
(
  const DVec&   xi,   // [in]
  const DVec&   yi,   // [in]
        double  ti,   // [in]
        DMat&   Qo    // [out]
)
//---------------------------------------------------------
{
  // function Q = ForwardStepIC2D(x2d, y2d, time)
  //
  // Purpose: compute plane flow configuration 

  double rho,u,v,p, rhou,rhov,Ener;

  rho = gamma; 
  u   = 3.0;
  v   = 0.0;
  p   = 1.0;

  // M = |u|/c,  c = sqrt(gamma*p/rho)
  rhou = rho * u;
  rhov = rho * v;
  Ener = p/gm1 + (SQ(rhou)+SQ(rhov))/(2.0*rho);

  assert(Qo.num_rows() == Np*K);
  Qo(All,1) = rho;
  Qo(All,2) = rhou;
  Qo(All,3) = rhov;
  Qo(All,4) = Ener; 
}
